### DISCUSSION TITLE
Usage of src/visualization/window[.h,.cpp]

### DISCUSSION THREAD

**Discussion by delhoume** [pmp-library/pmp-library#202] [General]
Hi,

I like very much this class, as it allows to write a complete clean app  similar (in concept to mpview).
It is very well written, and can be used a a basis for many  ImGui application of the "viewer" or "inspector"

I have used it for a tool that allows gigantic TIFF (I have one that is 1.5To) files to be inspected and previewed quickly, loading only what is visible.

<img width="1348" alt="image" src="https://github.com/pmp-library/pmp-library/assets/1763447/d0fd76ab-72b8-48b6-9c9b-4a813fd5996c">

The rendering of the tiles is done using only ImGui primitives

Can I distribute this class (slightly changed to remove dependencies to pmp)  ? in a public GitHub repo 
I would give credit to your work of course.

Thanks,

**Comment by dsieger** [comment]
Hi,

Sure you can create a derived version of the class and put it up on GitHub. Just make sure to respect the license terms and to maintain the original copyright information in the source files. Any acknowledgment would be appreciated of course.

The only somewhat tricky thing is the employee disclaimer I had to add to our license. I'm obviously not a lawyer and I can't give any legal advise on what this implies for derived works. However, if you just stick with the license I think you should be fine. 

Thanks,
